From 2646ed4937bb7dadee272f8bbc5da2b8f8df49ab Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin2.jansa@lgepartner.com>
Date: Wed, 19 Oct 2022 19:01:40 +0000
Subject: [PATCH] input.py: Remove usage of 'U' mode bit for opening files in
 python

The 'U' mode bit is removed in python 3.11. It has been
deprecated for a long time. The 'U' mode bit has no effect
so this change doesn't change any behavior.

See https://docs.python.org/3.11/whatsnew/3.11.html#changes-in-the-python-api
---
 gyp/pylib/gyp/input.py | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/gyp/pylib/gyp/input.py b/gyp/pylib/gyp/input.py
index d1742800..c30d6550 100644
--- a/gyp/pylib/gyp/input.py
+++ b/gyp/pylib/gyp/input.py
@@ -226,12 +226,7 @@ def LoadOneBuildFile(build_file_path, data, aux_data, includes,
     # Open the build file for read ('r') with universal-newlines mode ('U')
     # to make sure platform specific newlines ('\r\n' or '\r') are converted to '\n'
     # which otherwise will fail eval()
-    if sys.platform == 'zos':
-      # On z/OS, universal-newlines mode treats the file as an ascii file. But since
-      # node-gyp produces ebcdic files, do not use that mode.
-      build_file_contents = open(build_file_path, 'r').read()
-    else:
-      build_file_contents = open(build_file_path, 'rU').read()
+    build_file_contents = open(build_file_path, 'r').read()
   else:
     raise GypError("%s not found (cwd: %s)" % (build_file_path, os.getcwd()))
 
